examples/platforms/eagle/radio.c

```diff
@@ -0,0 +1,791 @@
+    {
+        otLogDebgPlat("State=OT_RADIO_STATE_DISABLED", NULL);
+        sState = OT_RADIO_STATE_DISABLED;
+    }
+
+    return OT_ERROR_NONE;
+}
+
+otError otPlatRadioSleep(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    otError error = OT_ERROR_INVALID_STATE;
+
+    if (sState == OT_RADIO_STATE_SLEEP || sState == OT_RADIO_STATE_RECEIVE)
+    {
+        otLogDebgPlat("State=OT_RADIO_STATE_SLEEP", NULL);
+        error  = OT_ERROR_NONE;
+        sState = OT_RADIO_STATE_SLEEP;
+        rf_set_txmode();
+    }
+
+    return error;
+}
+
+void rf_set_channel(uint8_t channel)
+{
+    current_channel = channel;
+    rf_set_chn(LOGICCHANNEL_TO_PHYSICAL(channel));
+}
+
+otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    otError error = OT_ERROR_INVALID_STATE;
+
+    if (sState != OT_RADIO_STATE_DISABLED)
+    {
+        otLogDebgPlat("State=OT_RADIO_STATE_RECEIVE", NULL);
+
+        error  = OT_ERROR_NONE;
+        sState = OT_RADIO_STATE_RECEIVE;
+        rf_set_channel(aChannel);
+        rf_set_rxmode();
+    }
+
+    return error;
+}
+
+/**
+ * Get the radio transmit frame buffer.
+ *
+ * OpenThread forms the IEEE 802.15.4 frame in this buffer then calls `otPlatRadioTransmit()` to request transmission.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @returns A pointer to the transmit frame buffer.
+ *
+ */
+otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    return &sTransmitFrame;
+}
+
+/**
+ * Get the most recent RSSI measurement.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @returns The RSSI in dBm when it is valid.  127 when RSSI is invalid.
+ *
+ */
+int8_t otPlatRadioGetRssi(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    signed char rssi_cur = -110;
+    rssi_cur             = rf_get_rssi();
+    return rssi_cur;
+}
+
+/**
+ * Begin the energy scan sequence on the radio.
+ *
+ * This function is used when radio provides OT_RADIO_CAPS_ENERGY_SCAN capability.
+ *
+ * @param[in] aInstance      The OpenThread instance structure.
+ * @param[in] aScanChannel   The channel to perform the energy scan on.
+ * @param[in] aScanDuration  The duration, in milliseconds, for the channel to be scanned.
+ *
+ * @retval OT_ERROR_NONE             Successfully started scanning the channel.
+ * @retval OT_ERROR_NOT_IMPLEMENTED  The radio doesn't support energy scanning.
+ *
+ */
+otError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, uint16_t aScanDuration)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(aScanChannel);
+    OT_UNUSED_VARIABLE(aScanDuration);
+
+    return OT_ERROR_NOT_IMPLEMENTED;
+}
+
+void otPlatRadioEnableSrcMatch(otInstance *aInstance, bool aEnable)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    sSrcMatchEnabled = aEnable;
+}
+
+static uint32_t m_in_critical_region = 0;
```

Comment:
Could we move this variable to the header of this file and rename it to `sInCriticalRegion`.